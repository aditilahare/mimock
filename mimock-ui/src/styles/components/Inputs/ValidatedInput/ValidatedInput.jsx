import React from 'react';
import PropTypes from 'prop-types';
import { InputBlock, FormInput, ErrorMessage } from './ValidatedInput.style';

function ValidatedInput({
	name,
	type,
	dataTestId,
	placeHolder,
	error,
	errorMessage,
	register,
}) {
	return (
		<InputBlock>
			<FormInput
				type={type}
				data-testid={dataTestId}
				placeholder={placeHolder}
				$error={error}
				{...register}
			/>
			<If condition={error}>
				<ErrorMessage data-testid={`input-error-${name}`}>
					{errorMessage}
				</ErrorMessage>
			</If>
		</InputBlock>
	);
}

ValidatedInput.propTypes = {
	name: PropTypes.string.isRequired,
	type: PropTypes.string,
	dataTestId: PropTypes.string.isRequired,
	placeHolder: PropTypes.string.isRequired,
	error: PropTypes.bool,
	errorMessage: PropTypes.string,
	register: PropTypes.object.isRequired,
};

ValidatedInput.defaultProps = {
	type: 'text',
	error: false,
	errorMessage: '',
};

export default ValidatedInput;
