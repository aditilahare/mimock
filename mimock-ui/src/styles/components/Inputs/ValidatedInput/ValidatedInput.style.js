import tw from 'tailwind-styled-components';

export const InputBlock = tw.div`
	block
	h-12
	min-h-12
	max-h-12
`;

export const FormInput = tw.input`
	block
	px-3
	py-1.5
	text-base
	font-normal
	text-gray-700
	bg-zinc-50
	bg-clip-padding
	border
	border-solid
	rounded
	mx-4
	focus:outline-blue-200
	${(props) => (props.$error ? 'border-2 border-rose-300' : 'border-gray-300')}
`;

export const ErrorMessage = tw.div`
	text-red-500
	text-xs
	font-sans
	subpixel-antialiased
	mx-4
	w-40
	absolute
`;
