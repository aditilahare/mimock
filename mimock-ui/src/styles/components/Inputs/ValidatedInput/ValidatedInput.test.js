import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';
import ValidatedInput from './ValidatedInput';

describe('ValidatedInput', () => {
	it('should render text input without error', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ValidatedInput
					name='text'
					dataTestId='text-input'
					placeHolder='Enter text'
					register={{}}
				/>
			);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(getByTestId('text-input')).toBeInTheDocument();
		expect(queryByTestId('input-error-text')).not.toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should render password input without error', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ValidatedInput
					type='password'
					name='password'
					dataTestId='password-input'
					placeHolder='Enter password'
					register={{}}
				/>
			);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(getByTestId('password-input')).toBeInTheDocument();
		expect(queryByTestId('input-error-password')).not.toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should render text input with error', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ValidatedInput
					name='text'
					dataTestId='text-input'
					placeHolder='Enter text'
					register={{}}
					error={true}
					errorMessage='Text is invalid'
				/>
			);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('text-input')).toBeInTheDocument();
		expect(getByTestId('input-error-text')).toHaveTextContent(
			'Text is invalid'
		);

		expect(container).toMatchSnapshot();
	});

	it('should accept input on triggering change event', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ValidatedInput
					name='text'
					dataTestId='text-input'
					placeHolder='Enter text'
					register={{}}
				/>
			);
		});

		const { getByTestId, queryByTestId, container } = tree;

		await act(async () => {
			fireEvent.change(getByTestId('text-input'), {
				target: { value: 'test' },
			});
		});

		expect(getByTestId('text-input')).toHaveValue('test');
		expect(queryByTestId('input-error-text')).not.toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
