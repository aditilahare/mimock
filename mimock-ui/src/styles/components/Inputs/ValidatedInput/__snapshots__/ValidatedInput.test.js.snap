// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidatedInput should accept input on triggering change event 1`] = `
<div>
  <div
    class="block h-12 min-h-12 max-h-12"
  >
    <input
      class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
      data-testid="text-input"
      placeholder="Enter text"
      type="text"
    />
  </div>
</div>
`;

exports[`ValidatedInput should render password input without error 1`] = `
<div>
  <div
    class="block h-12 min-h-12 max-h-12"
  >
    <input
      class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
      data-testid="password-input"
      placeholder="Enter password"
      type="password"
    />
  </div>
</div>
`;

exports[`ValidatedInput should render text input with error 1`] = `
<div>
  <div
    class="block h-12 min-h-12 max-h-12"
  >
    <input
      class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-2 border-rose-300"
      data-testid="text-input"
      placeholder="Enter text"
      type="text"
    />
    <div
      class="text-red-500 text-xs font-sans subpixel-antialiased mx-4 w-64 absolute"
      data-testid="input-error-text"
    >
      Text is invalid
    </div>
  </div>
</div>
`;

exports[`ValidatedInput should render text input without error 1`] = `
<div>
  <div
    class="block h-12 min-h-12 max-h-12"
  >
    <input
      class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
      data-testid="text-input"
      placeholder="Enter text"
      type="text"
    />
  </div>
</div>
`;
